draft-ietf-jsonpath-base.md

```diff
@@ -1026,12 +1026,15 @@ exist-expr        = [logical-not-op S] singular-path  ; path existence or non-ex
 
 Paths in filter expressions are Singular Paths, each of which selects at most one node.
 
+The current node is accessible via the current node identifier `@`.
+
 ~~~~ abnf
-singular-path     = rel-singular-path / abs-singular-path
-rel-singular-path = "@" *(S (name-segment / index-segment))
-abs-singular-path = root-identifier *(S (name-segment / index-segment))
-name-segment      = "[" name-selector "]" / dot-member-name-shorthand
-index-segment     = "[" index-selector "]"
+singular-path           = rel-singular-path / abs-singular-path
+rel-singular-path       = current-node-identifier *(S (name-segment / index-segment))
+current-node-identifier = "@"
+abs-singular-path       = root-identifier *(S (name-segment / index-segment))
```

Comment:
I would probably factor `*(S (name-segment / index-segment))` into a new rule `singular-path-segments`.  Not very strongly related to this PR though.

Reply:
I like it. Done.